File: chapter_03/3_6.py
K = 10
N = 20

count = 0
for x in range(K):
    for y in range(K):
        z = N-x-y
        if (z>=0) and (z<=K):
            count += 1

print(f"count : {count}")

"""
計算量はO(N^2)
"""

File: chapter_03/c++/3_6.cpp
#include<string>
#include<iostream>

using namespace std;

int main(){
    int K = 10, N = 8;

    int count = 0;

    for (int x=0; x<K; x++){
        for (int y=0; y<K; y++){
            int z = N - x - y;
            if (z >= 0 && z <= K) count ++;
        }
    }

    cout << "count : " << count << endl;

    /*
    計算量は0(N)
    */
}